includes/class-wc-gateway-komoju-block.php

```diff
@@ -45,6 +45,10 @@ public function get_payment_method_script_handles()
 
     public function get_payment_method_data()
     {
+        if (!is_null(WC()->cart) && WC()->cart->is_empty()) {
```

Comment:
Maybe `is_null(WC()->cart) || WC()->cart->is_empty()`? Two reasons:

* Customer reports indicate that the cart is null i.e., `read property "total" on null` at [this line](https://github.com/woocommerce/woocommerce/blob/9.4.3/plugins/woocommerce/includes/abstracts/abstract-wc-payment-gateway.php#L303)) 
* `is_null(WC()->cart) || ` makes sure that `WC()->cart` is _not_ null when we call `WC()->cart->is_empty()`

Reply:
Oh, when it is null I don't have to check it. Common null check mistake I guess. I will fix this after having lunch :D Thank you for pointing this out.

Reply:
Okay, there was a problem changing this. I need to check this is `is_admin()` or not. If I don't check that,
will see the error below. Because in admin edit mode, the card is null 😇 

<img width="1512" alt="Screenshot 2024-12-13 at 13 46 55" src="https://github.com/user-attachments/assets/132af33c-0280-4c11-b431-776a15a63b79" />

Reply:
@makotom Please check this change and press resolve conversation :D If you are okay with this.

Reply:
@Dinwy What if we move the check to L54?

What we want to prevent ultimately is creation of a session, and it's achievable by tweaking the `if` condition in L54. Then `get_payment_method_data` can return a "bogus" payment method information with null session when the cart is not available, which should be sufficient to make the checkout page editor functional, _I guess_.

It's worth to mention that, based on my interpretation of customer reports (I mean - it's still unclear), the error can happen even inside admin pages (i.e., some page editors), with which admin pages can break heavily, and therefore we might want to run the cart check under certain admin pages.

Reply:
In my opinion, L54 is different from what we've discussed here.

1. The first if statements check wether we have to create session or not.
2. L54 is checking our static variable is checking wether session is already created or not.

The below shows we are not making session for each payment method, but making one session and keep using it with different payment methods.

```php
        $logger = new WC_Logger();
        // We lazily fetch one session to be shared by all payment methods with dynamic fields.
        static $checkout_session;
        if (is_null($checkout_session)) {
            $logger->info('Creating session for fields' . ' ' . $this->name);
            $checkout_session = $this->gateway->create_session_for_fields();
        } else {
            $logger->info('already have session: ' . $this->name);
        }
```

```bash
2024-12-13T05:24:38+00:00 INFO Creating session for fields komoju_credit_card
2024-12-13T05:24:38+00:00 INFO already have session: komoju_paypay
2024-12-13T05:24:38+00:00 INFO already have session: komoju_linepay
2024-12-13T05:24:38+00:00 INFO already have session: komoju_konbini
2024-12-13T05:24:38+00:00 INFO already have session: komoju_bank_transfer
2024-12-13T05:24:38+00:00 INFO already have session: komoju_pay_easy
2024-12-13T05:24:38+00:00 INFO already have session: komoju_web_money
2024-12-13T05:24:38+00:00 INFO already have session: komoju_bit_cash
2024-12-13T05:24:38+00:00 INFO already have session: komoju_net_cash
2024-12-13T05:24:38+00:00 INFO already have session: komoju_credit_card
2024-12-13T05:24:38+00:00 INFO already have session: komoju_paypay
2024-12-13T05:24:38+00:00 INFO already have session: komoju_linepay
2024-12-13T05:24:38+00:00 INFO already have session: komoju_konbini
2024-12-13T05:24:38+00:00 INFO already have session: komoju_bank_transfer
2024-12-13T05:24:38+00:00 INFO already have session: komoju_pay_easy
2024-12-13T05:24:38+00:00 INFO already have session: komoju_web_money
2024-12-13T05:24:38+00:00 INFO already have session: komoju_bit_cash
2024-12-13T05:24:38+00:00 INFO already have session: komoju_net_cash
2024-12-13T05:24:38+00:00 INFO already have session: komoju_credit_card
2024-12-13T05:24:38+00:00 INFO already have session: komoju_paypay
2024-12-13T05:24:38+00:00 INFO already have session: komoju_linepay
2024-12-13T05:24:38+00:00 INFO already have session: komoju_konbini
2024-12-13T05:24:38+00:00 INFO already have session: komoju_bank_transfer
2024-12-13T05:24:38+00:00 INFO already have session: komoju_pay_easy
2024-12-13T05:24:38+00:00 INFO already have session: komoju_web_money
2024-12-13T05:24:38+00:00 INFO already have session: komoju_bit_cash
2024-12-13T05:24:38+00:00 INFO already have session: komoju_net_cash
```

I also added a fixed for the issue seeing total is null message by getting feedback from you. This is enough change for one small PR and I don't think we have to tweak this further. I am happy that we are not creating session anymore when cart is null and empty & in admin page.

**This is the error message we were getting. And fixed with `WC()->cart()` check.
> Warning: Attempt to read property "total" on null in /var/www/html/wp-content/plugins/woocommerce/includes/abstracts/abstract-wc-payment-gateway.php on line 303

Reply:
@makotom Another thing, here. FYI, current e2e master branch is broken. I opened PR for that. https://github.com/degica/komoju-e2e-tests/pull/180

Reply:
@Dinwy I looked at your latest commit to `create_session_for_fields` and realized that it's exactly what we needed, thanks! I agree it's sufficient now.

Let's hope the E2E test gets fixed soon...